core/src/main/java/com/google/zxing/datamatrix/encoder/MinimalEncoder.java

```diff
@@ -0,0 +1,1395 @@
+ * limitations under the License.
+ */
+
+package com.google.zxing.datamatrix.encoder;
+
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.zxing.common.CharacterSetECI;
+
+/**
+ * Encoder that encodes minimally
+ *
+ * Algorithm:
+ *
+ * Uses Dijkstra to produce mathematically minimal encodings that are in some cases smaller than the results produced
+ * by the algorithm described in annex S in the specification ISO/IEC 16022:200(E). The biggest improvment of this
+ * algorithm over that one is the case when the algorithm enters the most inefficient mode, the B256 mode. The 
+ * algorithm from the specification algorithm will exit this mode only if it encounters digits so that arbitrarily
+ * inefficient results can be produced if the postfix contains no digits.
+ *
+ * Multi ECI support and ECI switching:
+ *
+ * For multi language content the algorithm selects the most compact representation using ECI modes. Note that unlike
+ * the compaction algorithm used for QR-Codes, this implementation operates in two stages and therfore is not
+ * mathematically optimal. In the first stage, the input string is encoded minimally as a stream of ECI character set
+ * selectors and bytes encoded in the selected encoding. In this stage the algorithm might for example decide to
+ * encode ocurrences of the characters "\u0150\u015C" (O-double-acute, S-circumflex) in UTF-8 by a single ECI or
+ * alternatively by multiple ECIs that switch between IS0-8859-2 and ISO-8859-3 (e.g. in the case that the input
+ * contains many * characters from ISO-8859-2 (Latin 2) and few from ISO-8859-3 (Latin 3)).
+ * In a second stage this stream of ECIs and bytes is minimally encoded using the various Data Matrix encoding modes.
+ * While both stages encode mathematically minimally it is not ensured that the result is mathematically minimal since
+ * the size growth for inserting an ECI in the first stage can only be approximated as the first stage does not know 
+ * in which mode the ECI will occur in the second stage (may, or may not require an extra latch to ASCII depending on
+ * the current mode). The reason for this shortcoming are difficulties in implementing it in a straightforward and
+ * readable manner.
+ *
+ * GS1 support
+ *
+ * FNC1 delimiters can be encoded in the input string by using the FNC1 character specified in the encoding function.
+ * When a FNC1 character is specified then a leading FNC1 will be encoded and all ocurrences of delimiter characters
+ * while result in FNC1 codewords in the symbol.
+ *
+ * @author Alex Geller
+ */
+public final class MinimalEncoder {
+
+  enum Mode {
+    ASCII,
+    C40,
+    TEXT,
+    X12,
+    EDF,
+    B256
+  }
+
+
+  private MinimalEncoder() {
+    assert false;
+  }
+
+  static boolean isExtendedASCII(char ch, int fnc1) {
+
```

Comment:
Fix up the code style warnings noted in the automated checks and things like removing a blank line here

Reply:
I fixed that line but I failed to find the spot in GitHub where the automated check findings are displayed. There are 3 checks for Java down below (CodeQL Analyze, LGTM and Code scanning results / CodeQL). Is it one of them? I clicked around in the "details" of these but couldn't find any warnings. Can you point me to it?